cmd/podman/common/create.go

```diff
@@ -7,661 +7,713 @@ import (
+			&cf.CapAdd,
+			capAddFlagName, []string{},
+			"Add capabilities to the container",
+		)
+		_ = cmd.RegisterFlagCompletionFunc(capAddFlagName, completion.AutocompleteCapabilities)
 
-	capDropFlagName := "cap-drop"
-	createFlags.StringSliceVar(
-		&cf.CapDrop,
-		capDropFlagName, []string{},
-		"Drop capabilities from the container",
-	)
-	_ = cmd.RegisterFlagCompletionFunc(capDropFlagName, completion.AutocompleteCapabilities)
+		capDropFlagName := "cap-drop"
+		createFlags.StringSliceVar(
+			&cf.CapDrop,
+			capDropFlagName, []string{},
+			"Drop capabilities from the container",
+		)
+		_ = cmd.RegisterFlagCompletionFunc(capDropFlagName, completion.AutocompleteCapabilities)
 
-	cgroupnsFlagName := "cgroupns"
-	createFlags.String(
-		cgroupnsFlagName, "",
-		"cgroup namespace to use",
-	)
-	_ = cmd.RegisterFlagCompletionFunc(cgroupnsFlagName, AutocompleteNamespace)
+		cgroupnsFlagName := "cgroupns"
+		createFlags.String(
+			cgroupnsFlagName, "",
+			"cgroup namespace to use",
+		)
+		_ = cmd.RegisterFlagCompletionFunc(cgroupnsFlagName, AutocompleteNamespace)
 
-	cgroupsFlagName := "cgroups"
-	createFlags.StringVar(
-		&cf.CGroupsMode,
-		cgroupsFlagName, cgroupConfig(),
-		`control container cgroup configuration ("enabled"|"disabled"|"no-conmon"|"split")`,
-	)
-	_ = cmd.RegisterFlagCompletionFunc(cgroupsFlagName, AutocompleteCgroupMode)
+		cgroupsFlagName := "cgroups"
+		createFlags.StringVar(
+			&cf.CGroupsMode,
+			cgroupsFlagName, cgroupConfig(),
+			`control container cgroup configuration ("enabled"|"disabled"|"no-conmon"|"split")`,
+		)
+		_ = cmd.RegisterFlagCompletionFunc(cgroupsFlagName, AutocompleteCgroupMode)
 
-	cgroupParentFlagName := "cgroup-parent"
-	createFlags.StringVar(
-		&cf.CGroupParent,
-		cgroupParentFlagName, "",
-		"Optional parent cgroup for the container",
-	)
-	_ = cmd.RegisterFlagCompletionFunc(cgroupParentFlagName, completion.AutocompleteDefault)
+		cpuPeriodFlagName := "cpu-period"
+		createFlags.Uint64Var(
+			&cf.CPUPeriod,
+			cpuPeriodFlagName, 0,
+			"Limit the CPU CFS (Completely Fair Scheduler) period",
+		)
+		_ = cmd.RegisterFlagCompletionFunc(cpuPeriodFlagName, completion.AutocompleteNone)
 
-	cidfileFlagName := "cidfile"
-	createFlags.StringVar(
-		&cf.CIDFile,
-		cidfileFlagName, "",
-		"Write the container ID to the file",
-	)
-	_ = cmd.RegisterFlagCompletionFunc(cidfileFlagName, completion.AutocompleteDefault)
+		cpuQuotaFlagName := "cpu-quota"
+		createFlags.Int64Var(
+			&cf.CPUQuota,
+			cpuQuotaFlagName, 0,
+			"Limit the CPU CFS (Completely Fair Scheduler) quota",
+		)
+		_ = cmd.RegisterFlagCompletionFunc(cpuQuotaFlagName, completion.AutocompleteNone)
 
-	conmonPidfileFlagName := "conmon-pidfile"
-	createFlags.StringVar(
-		&cf.ConmonPIDFile,
-		conmonPidfileFlagName, "",
-		"Path to the file that will receive the PID of conmon",
-	)
-	_ = cmd.RegisterFlagCompletionFunc(conmonPidfileFlagName, completion.AutocompleteDefault)
+		cpuRtPeriodFlagName := "cpu-rt-period"
```

Comment:
Potential issue: We don't want these (resource limit flags) to be passed into the infra container - they need to be handled at the pod level, not the infra container level. I think right now your code will pass them into the infra container?